node/network/statement-distribution/src/lib.rs

```diff
@@ -1928,25 +1970,50 @@ struct MetricsInner {
 	active_leaves_update: prometheus::Histogram,
 	share: prometheus::Histogram,
 	network_bridge_update_v1: prometheus::Histogram,
+	statements_unexpected: prometheus::CounterVec<prometheus::U64>,
+}
+
+// Used to distinguish different unexpected statements in the metrics
+enum UnexpectedStatementLabel {
+	Seconded,
+	Valid,
+	Large,
+}
+
+impl From<UnexpectedStatementLabel> for &'static str {
+	fn from(label: UnexpectedStatementLabel) -> Self {
+		match label {
+			UnexpectedStatementLabel::Seconded => "seconded",
+			UnexpectedStatementLabel::Valid => "valid",
+			UnexpectedStatementLabel::Large => "large",
+		}
+	}
```

Comment:
nit: Commonly implementing `std::fmt::Display` is the preferred way.

Reply:
actually in this specific case, we use each variant only once, 3 constants or just string literals would be even better

Reply:
Yes, I agree - I planned to use more variants but even 3 is redundant as we actually see only `valid` label all the time. But I have converted that to constants.